src/core/providers/qgsprovidermetadata.h

```diff
@@ -324,6 +326,22 @@ class CORE_EXPORT QgsProviderMetadata : public QObject
      */
     virtual bool uriIsBlocklisted( const QString &uri ) const;
 
+    /**
+     * Queries the specified \a uri and returns a list of any valid sublayers found in the dataset which can be handled by this provider.
+     *
+     * The optional \a flags argument can be used to control the behavior of the query.
+     *
+     * The optional \a feedback argument can be used to provide cancellation support for long-running queries.
+     *
+     * \note Providers which implement this method should return always return a list of sublayer details for any valid, even if the \a uri
```

Comment:
```suggestion
     * \note Providers which implement this method should always return a list of sublayer details for any valid, even if the \a uri
```

Not entirely sure about the wording here.